src/routes/auth/magic-link.ts

```diff
@@ -0,0 +1,95 @@
+import { APPLICATION } from '@shared/config'
+import { Request, Response } from 'express'
+import Boom from '@hapi/boom'
+import { accountOfRefreshToken, activateAccount } from '@shared/queries'
+import { asyncWrapper } from '@shared/helpers'
+import { request } from '@shared/request'
+import { v4 as uuidv4 } from 'uuid'
+import { magicLinkQuery } from '@shared/validation'
+import { AccountData, Session, UpdateAccountData, UserData } from '@shared/types'
+import { createHasuraJwt, newJwtExpiry } from '@shared/jwt'
+import { setRefreshToken } from '@shared/cookies'
+
+async function magicLink({ query }: Request, res: Response): Promise<unknown> {
+  const { token, action } = await magicLinkQuery.validateAsync(query);
+
+  const useCookie = typeof query.cookie !== 'undefined' ? query.cookie === 'true' : true
+
+  let refresh_token = token;
+
+  if (action === 'sign-up') {
```

Comment:
I'm thinking we want to remove this dependency. And possibly remove the possibility to "register" with a magic link.

I think the magic link should only be accessible on `.login()` and not on `register()`.

HBP should then check if the account exists or not during login, and create the account if it does not exist.

Reply:
We agreed to use register/login independantly.